### ISSUE TITLE
Buggy Timecode calculation when invoked with `frames`

### ISSUE THREAD

**Issue by rsomani95** [eoyilmaz/timecode#30]
When creating a `Timecode` object as such:
```python
framerate = Fraction(24000, 1001) #23.976023976023978
frame_idx = 50000

Timecode(framerate, frames=frame_idx)
```

gives the output `00:36:13:20` which is wrong. I've manually verified this with the input video and also checked the result with 3 other Timecode calculators:
1. [Omnicalculator](https://www.omnicalculator.com/other/frames-to-timecode)
2. [Zapstudio](https://www.zapstudio.net/framecalc/)
3. PySceneDetect's [FrameTimecode](https://github.com/Breakthrough/PySceneDetect/blob/master/scenedetect/frame_timecode.py) module

All of these above output `00:34:45:09` (except PySceneDetect which outputs in milliseconds, but still accurate), which is the correct output

**Comment by eoyilmaz** [comment]
Hey @rsomani95 thank you for your bug report. I'll check it as soon as I have time.

**Comment by eoyilmaz** [comment]
So I had some free time to invest on this bug report.

Firstly, the usage of ``Fraction`` module was not supported. I've updated the code to handle that. So it is now possible to supply ``Fraction`` for frame rate and I'll create a new release for that.

But either that way what I get as a result is ``00:34:43:07`` unless I set ``Timecode.drop_frame`` to ``True`` which then returns ``00:34:45;21``. And I believe ``23.976`` is not a drop frame frame rate by default.

Also I checked with DaVinci Resolve, to test this, I have created a new Timeline with 23.976 as the frame rate (DaVinci Resolve doesn't allow me to set the timeline to drop frame), created a Title clip extended it so the duration is 50000 frames and the Timecode at the end of the clip (the last frame of the clip) was ``00:34:43:07``. So I proved that the calculated Timecode is correct.

Fixed in 0ad0c59.
